Source/View Controllers/UploadViewController.swift

```diff
@@ -230,22 +245,26 @@ class UploadViewController: NSViewController {
     }
 
     func updateCredentialsAvailable() {
-        guard credentialsAvailable != (
-            !jamfProServerLabel.stringValue.isEmpty
-            && (username != nil)
-            && !username.isEmpty
-            && (password != nil)
-            && !password.isEmpty) else { return }
-        credentialsAvailable = !credentialsAvailable
+        guard let username = username, !username.isEmpty,
+              let password = password, !password.isEmpty,
+              !jamfProServerLabel.stringValue.isEmpty else {
+            if credentialsAvailable { credentialsAvailable = false }
+            return
+        }
+
+        if !credentialsAvailable { credentialsAvailable = true }
```

Comment:
This function implementation looks overly complicated and hard to understand.  How about this instead?

```
if let username = username, !username.isEmpty,
   let password = password, !password.isEmpty,
   !jamfProServerLabel.stringValue.isEmpty {
   credentialsAvailable = true
} else {
   credentialsAvailable = false
}
```

Reply:
for me it's absolutely fine. Just done it the complicated way, to not change the behaviour of not changing the variable in case it has the same value.